glass/observations.py

```diff
@@ -328,3 +330,192 @@ def tomo_nz_gausserr(
     binned_nz *= nz
 
     return binned_nz  # type: ignore[no-any-return]
+
+
+class angular_variable_depth_mask:
+    r"""Variable depth mask for tomographic bins.
+
+    This class allows to create a mask with a different variable depth mask in the angular direction for each tomographic bin.
+
+    Parameters
+    ----------
+    vardepth_map : array_like
+        Map of variable which traces the depth per tomographic bin.
+    n_bins : int
+        Number of tomographic bins.
+    zbins : array_like
+        Shell redshift limits.
+    """
+
+    def __init__(self, vardepth_map, n_bins, zbins):
+        self.vardepth_map = vardepth_map
+        self.n_bins = n_bins
+        self.zbins = zbins
+
+    def test_index(self, index):
+        r"""Test the index for validity."""
+        if not isinstance(index, tuple):
+            raise ValueError("Index must be an tuple of two integers")
+
+        if index[0] >= self.n_bins:
+            raise ValueError("Leading index cannot exceed number of tomographic bins")
+
+        if index[1] >= len(self.zbins):
+            raise ValueError("Trailing index cannot exceed number of shells")
+
+    def __getitem__(self, index):
+        r"""Get the mask for the given index.
+
+        Parameters
+        ----------
+        index : tuple
+            Indices of the tomographic bin and shell pair.
+
+        Returns
+        -------
+        mask : array_like
+            Mask for the given index.
+
+        Raises
+        ------
+        ValueError
+            If the index is invalid.
+
+        """
+        self.test_index(index)
+
+        return self.vardepth_map[index[0]]
+
+
+class angular_los_variable_depth_mask(angular_variable_depth_mask):
```

Comment:
Again following `N801`

```suggestion
class AngularLosVariableDepthMask(AngularVariableDepthMask):
```